### PR TITLE
buttonBorderUpdate, needed for urbn border for UIOS-536

### PR DESCRIPTION
**Pull request by bhtran**
The purpose of this PR is to add a way to allow the user to define the urbn_topBorder on the buttonSV to match the comp.
Parent Ticket: [UIOS-82](https://urbnit.atlassian.net/browse/UIOS-82)
Sub Ticket: [UIOS-536](https://urbnit.atlassian.net/browse/UIOS-536) 
Comp: https://app.zeplin.io/project/596cd57542a7632c71b62b8a/screen/596cd5fd5c81c2ec4bf2580b

### CODE REVIEW

File: URBNSwiftAlert/Classes/AlertStyler.swift [hunk 1]
```diff
@@ -73,6 +73,16 @@ extension AlertStyler {
...
+
+        /**
+         * Separator color between the title/message and button, uses urbn_topBorder on buttonSV
+         */
+        public var separatorColor: UIColor = .clear
+
+        /**
+         * Separator width betwenn the title/message and button, uses urbn_topBorder on buttonSV
+         */
+        public var separatorWidth: Int = 0
```
Comment (codecaffeine) [inline URBNSwiftAlert/Classes/AlertStyler.swift]: Is this pixel or point? Should specify in the variable name

**Review by codecaffeine — CHANGES_REQUESTED** [review]

File: URBNSwiftAlert/Classes/AlertStyler.swift (hunk 1 above)
Comment (ryang1428) [reply URBNSwiftAlert/Classes/AlertStyler.swift]: I would be a bit more explicit with the naming of these. Maybe `buttonContainerSeperator....`.

**Review by ryang1428 — CHANGES_REQUESTED** [review]

File: URBNSwiftAlert/Classes/AlertStyler.swift (hunk 1 above)
Comment (bhtran) [reply URBNSwiftAlert/Classes/AlertStyler.swift]: Pixel, will update

**Review by KVTaniguchi — CHANGES_REQUESTED** [review]
update one of the examples in ExampleViewController to demo this new feature?

**Comment by bhtran** [comment]
@KVTaniguchi added the example on the example VC

**Comment by codecaffeine** [comment]
FYI: Most of these changes now are with pod file updates

**Review by codecaffeine — APPROVED** [review]

**Comment by KVTaniguchi** [comment]
@bhtran When I built this I wanted to avoid the "2 new properties for every 1 new UI element" trap URBNAlert was starting to get into, so in URBNSwiftyAlert I defined the ability for users to create their own custom views (with any styling they want).  So instead of introducing additional style properties, you can just add your own custom buttons -  see `All Custom Views`, `func showFullCustomAlert()` and `class ExampleCustomButtons: UIView, AlertButtonContainer {`

You should notice in this screenshot that the part of the alert that has the buttons has not only a border on the top of the buttons but between them as well. 

simulator screen shot - iphone x - 2017-11-07 at 21 17 39

**Review by KVTaniguchi — CHANGES_REQUESTED** [review]
no new properties please, just use custom views.  see `func showFullCustomAlert()` and `ExampleCustomButtons()` in the `ExampleViewController`

**Comment by ryang1428** [comment]
@KVTaniguchi The only thing here is that _every_ URBNAlert we display in WL will have this line-- so we wanted to avoid having to use a custom view for every alert. That being said, this could be simplified to 1 styler property by just making `buttonContainerSeparatorBorderStyle` instead of having the color & width separate

**Comment by KVTaniguchi** [comment]
@bhtran  please consolidate to buttonContainerSeparatorBorderStyle

**Review by KVTaniguchi — CHANGES_REQUESTED** [review]
please consolidate to buttonContainerSeparatorBorderStyle

**Comment by bhtran** [comment]
@KVTaniguchi ready to be looked at

File: Example/URBNSwiftAlert/ExampleViewController.swift [hunk 2]
```diff
@@ -41,8 +41,9 @@ class ExampleViewController: UIViewController {
         customStyleAlert.alertStyler.alertViewShadow.offset = CGSize(width: 5, height: 5)
         customStyleAlert.alertStyler.alertViewShadow.radius = 5.0
         customStyleAlert.alertStyler.alertViewShadow.opacity = 0.9
-        customStyleAlert.alertStyler.alert.buttonContainerSeparatorColor = .green
-        customStyleAlert.alertStyler.alert.buttonContainerSeparatorPixelWidth = 5
+//        customStyleAlert.alertStyler.alert.buttonContainerSeparatorColor = .green
+//        customStyleAlert.alertStyler.alert.buttonContainerSeparatorPixelWidth = 5
```
Comment (ryang1428) [inline Example/URBNSwiftAlert/ExampleViewController.swift]: remove this commented code please

**Comment by ryang1428** [comment]
@bhtran I tried to implement a `buttonContainerSeparatorBorderStyle` on the "One Button" and "Two Button" examples, and it is not displaying correctly:

i.e.
```
    func showOneButtonAlert() {
        let oneBtnAlert = AlertViewController(title: wrappingTitle, message: longMessage)
        oneBtnAlert.alertStyler.alert.buttonContainerSeparatorBorderStyle = BorderStyle(color: .red, pixelWidth: 5)

        oneBtnAlert.addActions(genericDoneAction)
        oneBtnAlert.show()
    }
```

I would use that as an example to test with as it is most like what we will be using in WL. You may have to tweak where you are setting that top border

**Comment by bhtran** [comment]
@ryang1428 The border is for the buttonSV only, but it seems that it only works when the button's background color is set to clear. But I think that wouldn't be the best way to implement a border but have to account for the bg color everytime.

**Review by ryang1428 — APPROVED** [review]
Cool with this now. Looking as expected on the one/two button examples as well

**Comment by codecaffeine** [comment]
I approve. @KVTaniguchi feel free to merge when you approve

**Review by KVTaniguchi — APPROVED** [review]

**Pull request merged**